### ISSUE TITLE
-Wsometimes-uninitialized in kernel/seccomp.c

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#1056]
aarch64 linux-next:
```
kernel/seccomp.c:1112:2: warning: variable 'knotif' is used uninitialized whenever 'for' loop exits because its condition is false [-Wsometimes-uninitialized]
        list_for_each_entry(cur, &filter->notif->notifications, list) {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./include/linux/list.h:602:7: note: expanded from macro 'list_for_each_entry'
             &pos->member != (head);                                    \
             ^~~~~~~~~~~~~~~~~~~~~~
kernel/seccomp.c:1124:7: note: uninitialized use occurs here
        if (!knotif) {
             ^~~~~~
kernel/seccomp.c:1112:2: note: remove the condition if it is always true
        list_for_each_entry(cur, &filter->notif->notifications, list) {
        ^
./include/linux/list.h:602:7: note: expanded from macro 'list_for_each_entry'
             &pos->member != (head);                                    \
             ^
kernel/seccomp.c:1094:31: note: initialize the variable 'knotif' to silence this warning
        struct seccomp_knotif *knotif, *cur;
                                     ^
                                      = NULL
```

**Comment by nathanchance** [comment]
I sent a report a little bit ago: https://lore.kernel.org/lkml/20200617200844.GA12976@Ryzen-9-3900X.localdomain/

**Comment by nickdesaulniers** [comment]
cc @kees 

thanks for reporting.

**Comment by kees** [comment]
This should be fixed in next-20200618.
